src/AAT.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @description PocketAAT entry point.
+ */
+
+import { sha3_256 } from 'js-sha3';
+import ed25519 = require('ed25519');
+
+export class PocketAAT {
+  public readonly version: string;
+  public readonly clientPublicKey: string;
+  public readonly applicationPublicKey: string;
+  private applicationSignature: string;
+
+  constructor(version: string, clientPublicKey: string, applicationPublicKey: string) {
+    this.version = version;
+    this.applicationSignature = '';
+    this.clientPublicKey = clientPublicKey;
+    this.applicationPublicKey = applicationPublicKey;
+
+    if (!this.isValid()) {
+      throw new TypeError('Invalid properties length.');
+    }
+  }
+
+  public sign(privateKey: string) {
```

Comment:
Is there a more secure way of passing the `privateKey` parameter than a string?

Reply:
At the point it became insecure, the backend running the AAT-SDK will be compromised, even if we create a buffer or let the developer read from a file, the method is public.